ui/src/app/machines/components/MachineHeaderForms/ActionFormWrapper/ActionFormWrapper.tsx

```diff
@@ -128,92 +122,89 @@ export const ActionFormWrapper = ({
   }, [machinesToAction, clearHeaderContent]);
 
   const getFormComponent = () => {
-    if (headerContent && headerContent.name) {
-      switch (headerContent.name) {
-        case NodeActions.CLONE:
-          return (
-            <CloneForm
-              actionDisabled={actionDisabled}
-              clearHeaderContent={clearHeaderContent}
-              setSearchFilter={setSearchFilter}
-              viewingDetails={viewingDetails}
-            />
-          );
-        case NodeActions.COMMISSION:
-          return (
-            <CommissionForm
-              actionDisabled={actionDisabled}
-              clearHeaderContent={clearHeaderContent}
-            />
-          );
-        case NodeActions.DEPLOY:
-          return (
-            <DeployForm
-              actionDisabled={actionDisabled}
-              clearHeaderContent={clearHeaderContent}
-            />
-          );
-        case NodeActions.MARK_BROKEN:
-          return (
-            <MarkBrokenForm
-              actionDisabled={actionDisabled}
-              clearHeaderContent={clearHeaderContent}
-            />
-          );
-        case NodeActions.OVERRIDE_FAILED_TESTING:
-          return (
-            <OverrideTestForm
-              actionDisabled={actionDisabled}
-              clearHeaderContent={clearHeaderContent}
-            />
-          );
-        case NodeActions.RELEASE:
-          return (
-            <ReleaseForm
-              actionDisabled={actionDisabled}
-              clearHeaderContent={clearHeaderContent}
-            />
-          );
-        case NodeActions.SET_POOL:
-          return (
-            <SetPoolForm
-              actionDisabled={actionDisabled}
-              clearHeaderContent={clearHeaderContent}
-            />
-          );
-        case NodeActions.SET_ZONE:
-          return (
-            <SetZoneForm
-              actionDisabled={actionDisabled}
-              clearHeaderContent={clearHeaderContent}
-            />
-          );
-        case NodeActions.TAG:
-          return (
-            <TagForm
-              actionDisabled={actionDisabled}
-              clearHeaderContent={clearHeaderContent}
-            />
-          );
-        case NodeActions.TEST:
-          return (
-            <TestForm
-              actionDisabled={actionDisabled}
-              clearHeaderContent={clearHeaderContent}
-              {...headerContent.extras}
-            />
-          );
-        default:
-          return (
-            <FieldlessForm
-              actionDisabled={actionDisabled}
-              headerContent={headerContent}
-              clearHeaderContent={clearHeaderContent}
-            />
-          );
-      }
+    switch (action) {
+      case NodeActions.CLONE:
```

Comment:
@huwshimi I think what I want to get at in `ActionFormWrapper` is that by the time it's rendered it's because an action is *definitely* selected, and everything in that component and its children only interact with `NodeActions`. It's the component above, `MachineHeaderForms` that tracks the other enum `MachineHeaderNames`, which determines whether to render an "action" or "non-action" form.

Adding `action` to `extras` isn't a bad idea though, I think I could probably get rid of that awkward util with that. But then we just have to remember to always pass `{ extras: { action: NodeActions.SOMETHING } }` when setting the header content. Anyway let me know if you want to jump on a call and maybe come up with something that's less confusing.

Reply:
Hmm after giving the `extras` thing a go I don't think I like it. As awkward as that util is, I think I prefer the flow where `headerContent.name` is the absolute controller of what gets rendered in the header. It determines what to show in the title and what form to render, whether that be an action or non-action form or whatever. But with `extras` now there are two things that determine what gets rendered, which I think could lead to confusion about what's shown and why.

I'll try your other suggestion of using the map in `FieldlessForm`